packages/@lwc/ssr-compiler/src/compile-js/generate-markup.ts

```diff
@@ -65,11 +66,14 @@ const bGenerateMarkup = esTemplate`
         yield* tmplFn(props, attrs, slotted, ${/*component class*/ 3}, instance);
         yield \`</\${tagName}>\`;
     }
-`<ExportNamedDeclaration>;
+    ${/* component class */ 3}[__SYMBOL__GENERATE_MARKUP] = generateMarkup;
+`<[ExportNamedDeclaration, ExpressionStatement]>;
 
-const bAssignGenerateMarkupToComponentClass = esTemplate`
-${/* lwcClassName */ is.identifier}[__SYMBOL__GENERATE_MARKUP] = generateMarkup;
-`<ExpressionStatement>;
+const bExposeTemplate = esTemplate`
+    if (${/*template*/ is.identifier}) {
+        ${/* component class */ is.identifier}[__SYMBOL__DEFAULT_TEMPLATE] = ${/*template*/ 0}
+    }
+`<IfStatement>;
```

Comment:
I tried using `isNullableOf` so that this could be part of `bGenerateMarkup`, but it didn't work, so I just kept it separate.